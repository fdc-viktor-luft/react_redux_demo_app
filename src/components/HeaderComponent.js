// @flow
import React from 'react';
import { PageHeader } from 'react-bootstrap';

export class HeaderComponent extends React.Component {
    render() {
        return <PageHeader>React + Redux Demo</PageHeader>;
    }
}
