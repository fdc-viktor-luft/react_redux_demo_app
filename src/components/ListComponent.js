// @flow
import React from 'react';
import { connect } from 'react-redux';
import { ListGroup, ListGroupItem } from 'react-bootstrap';

import type { StoreType } from '../state/StoreProvider';
import { Jokes } from "../business/Jokes";

export class DisconnectedListComponent extends React.Component {

    render() {
        return (
            <ListGroup fill>
                {
                    this.props.jokes.getJokes().map((value, index) => {
                        return (
                            <ListGroupItem key={'joke-' + index} className={'clearfix list-group-item-test-' + index}>
                                {value}
                            </ListGroupItem>
                        )
                    }).toArray()
                }
            </ListGroup>
        );
    }
}

DisconnectedListComponent.propTypes = {
    // injected through redux
    jokes: React.PropTypes.instanceOf(Jokes).isRequired
};

const mapStateToProps = (state: StoreType) => {
    return {
        jokes: state.jokes
    };
};

export const ListComponent = connect(mapStateToProps)(DisconnectedListComponent);

