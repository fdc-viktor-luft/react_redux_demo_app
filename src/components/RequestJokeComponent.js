// @flow
import React from 'react';
import { connect } from 'react-redux';
import { Panel, FormGroup,  Button, ControlLabel, FormControl } from 'react-bootstrap';
import { Actions } from '../state/Actions';

export class DisconnectedRequestJokeComponent extends React.Component {

    state: {
        input: string
    };

    constructor() {
        super();
        (this: any)._onInputChange = this._onInputChange.bind(this);
        (this: any)._onSubmit = this._onSubmit.bind(this);

        // setup initial state
        this.state = {
            input: ''
        }
    }

    _onInputChange(e: any) {
        this.setState({
            input: e.target.value
        });
    }

    _onSubmit() {
        const parsedInt = parseInt(this.state.input, 10);
        if (!isNaN(parsedInt)) {
            this.props.requestJoke(parsedInt);
            this.setState({
                input: ''
            });
        }
    }

    render() {
        return (
            <Panel>
                <FormGroup>
                    <ControlLabel>Request more Jokes!</ControlLabel>
                    <FormControl type="number"
                                 className="form-control-test"
                                 value={this.state.input}
                                 onChange={this._onInputChange}
                                 placeholder="Enter id"/>
                </FormGroup>

                <Button bsStyle="primary" onClick={this._onSubmit} className="test-submit-button">
                    Request joke with this id
                </Button>
            </Panel>
        );
    }
}

DisconnectedRequestJokeComponent.propTypes = {
    // injected through redux
    requestJoke: React.PropTypes.func.isRequired
};

const mapDispatchToProps = (dispatch: any) => {
    return {
        requestJoke: (num) => dispatch(Actions.requestJoke(num))
    };
};

export const RequestJokeComponent = connect(null, mapDispatchToProps)(DisconnectedRequestJokeComponent);
