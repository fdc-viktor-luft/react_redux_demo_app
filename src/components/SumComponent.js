// @flow
import React from 'react';
import { connect } from 'react-redux';
import { Well } from 'react-bootstrap';

import type { StoreType } from '../state/StoreProvider';
import { Jokes } from "../business/Jokes";

export class DisconnectedSumComponent extends React.Component {
    render() {
        const sum = this.props.jokes.getJokes().reduce((red, elem) => red + elem.length, 0);
        return (
            <Well className="test-class-sum">
                {'Sum of all characters: ' + sum }
            </Well>
        );
    }
}

DisconnectedSumComponent.propTypes = {
    // injected through redux
    jokes: React.PropTypes.instanceOf(Jokes).isRequired
};

export const SumComponent = connect(
    (state: StoreType) => {
        return {
            jokes: state.jokes
        };
    }
)(DisconnectedSumComponent);
