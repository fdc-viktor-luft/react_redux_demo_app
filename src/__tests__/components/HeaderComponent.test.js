import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import { HeaderComponent } from '../../components/HeaderComponent';

describe('HeaderComponent', () => {
    it('[snapshot] it renders correctly', () => {
        const app = shallow(<HeaderComponent />);
        expect(toJson(app)).toMatchSnapshot();
    });

});
