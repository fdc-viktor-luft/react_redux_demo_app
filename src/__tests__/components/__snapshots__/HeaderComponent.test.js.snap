exports[`HeaderComponent [snapshot] it renders correctly 1`] = `
<PageHeader
  bsClass="page-header">
  React + Redux Demo
</PageHeader>
`;
